lib/src/license_detection/primary_filter.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:pana/src/license_detection/license.dart';
+
+/// Filters the corpus licenses and
+/// returns a list of [License] which might
+/// be a possible match.
```

Comment:
This format for comments applies to all documentation comments, see effective dart.
```dart
/// <short single sentence explaining what this does>
///
/// <longer description if necessary>
```

In this case I would do something like:

```dart
/// Filters [knownLicenses] and returns a list of licenses that could likely be contained within [occurrences].
///
/// A license from set of [knownLicenses] is considered likely to be contained in within [occurrences] if.....
```